torchtitan/experiments/deepseek_v3/symm_mem_recipes/triton_on_device_all_to_all_v.py

```diff
@@ -156,3 +158,34 @@ def on_device_all_to_all_v(
     )
     # log_triton_kernel(kernel)
     return output
+
+
+class OnDeviceAllToAllV(torch.autograd.Function):
+    @staticmethod
+    def forward(
+        ctx,
+        output: torch.Tensor,
+        output_splits: torch.Tensor,
+        input: torch.Tensor,
+        input_splits: torch.Tensor,
+        group: dist.ProcessGroup = dist.group.WORLD,
+    ):
+        _on_device_all_to_all_v(output, output_splits, input, input_splits, group=group)
+        ctx.save_for_backward(output_splits)
+        ctx.group = group
+        ctx.input_shape = input.shape
+        return output
+
+    @staticmethod
+    def backward(ctx, grad_output):
+        grad_output_splits = ctx.saved_tensors
+        grad_input_splits = torch.empty_like(grad_output_splits)
+        grad_input = grad_output.new_empty(*ctx.input_shape)
+        _on_device_all_to_all_v(
+            grad_input, grad_input_splits, grad_output, grad_output_splits, group=group
```

Comment:
I might be missing something but I don't think this `backward()` function is ever being called -- otherwise `group` would fail here since it should be `ctx.group`

Reply:
yea good point. but it does look like the on_device all2allv is called in forward part of moe layer in a part that has gradients enabled.  (it is also called under 'with no_grad()` but not exclusively)

Reply:
@EugenHotaj @wconstab Thanks. You are right. 
I am backing out the backward function for `on_device_all_to_all_v` from this PR.
The reason is that autograd does not allow touching the buffer a second time between fwd and bwd. This conflicts with sharing the symm mem buffer across layers (to save memory) as in PR #958.

Instead, I am adding backward support via the traditional `all_to_all_single_autograd` (Yes, there is a CPU sync.) But the backward is more stable. PR is at #989.

`on_device_all_to_all_v` is now limited to inference / generation.

Reply:
sounds good for short term.  longer term we'd want to fix the backward+symm memm issue though right?  can we mark the tensor as a 'buffer' to make autograd not care about it, or something?

Reply:
Autograd does not allow touching the buffer that's going to be used by autograd, is that the case here? If you still need all this memory live, then registering multiple copies with symmetric memory shouldn't be overall memory hit.

Reply:
@wconstab @ngimel yes. 
@EugenHotaj and @yifuwang suggested something similar -- we can try moving the symm mem inside an autograd function, so that it is not visible to the autograd engine.